### ISSUE TITLE
Greenmask V0.1.13 SIGSEGV

### ISSUE THREAD

**Issue by TCY16** [GreenmaskIO/greenmask#102]
Hi!

While using Greenmask for the first time, I've encountered a segfault. Using the `dump` option while testing on a single table in my database. I've verified connection with `psql` following config:
```
dump:
  pg_dump_options:
    dbname: "host=obfuscated-amazon-address user=postgres dbname=db_name"
    jobs: 10
    table: "tablename"
storage:
  type: "directory"
  directory:
    path: "/home/ssm-user/tmp"
```

The segfault output is:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xd3ddb1]

goroutine 151 [running]:
github.com/jackc/pgx/v5.(*LargeObject).Close(...)
	/home/runner/go/pkg/mod/github.com/jackc/pgx/v5@v5.5.5/large_objects.go:156
github.com/greenmaskio/greenmask/internal/db/postgres/dumpers.(*BlobsDumper).Execute.func2.2()
	/home/runner/work/greenmask/greenmask/internal/db/postgres/dumpers/large_object.go:95 +0x31
github.com/greenmaskio/greenmask/internal/db/postgres/dumpers.(*BlobsDumper).Execute.func2()
	/home/runner/work/greenmask/greenmask/internal/db/postgres/dumpers/large_object.go:100 +0x1f1
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 65
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x96
```

For completeness; The `uname -r` output gives: `6.1.79-99.167.amzn2023.x86_64`. I've used `greenmask-linux-amd64.tar.gz` and the following libraries `postgresql15 postgresql15-contrib`. The database version is Postgres 15 as well.

On a different note, I've also found that `greenmask dump --data-only` gives a schema collision error : `"pg_dump: error: options -s/--schema-only and -a/--data-only cannot be used together"`

**Comment by wwoytenko** [comment]
Hi! Thank you for reporting!

> github.com/greenmaskio/greenmask/internal/db/postgres/dumpers.(*BlobsDumper).Execute.func2.2()

Looks like the seg fault is in Large Object dumping. I will check this tomorrow. There is definitely a bug.

> On a different note, I've also found that greenmask dump --data-only gives a schema collision error : "pg_dump: error: options -s/--schema-only and -a/--data-only cannot be used together"

Well, you're right. I will review this case tomorrow as well.

**Comment by wwoytenko** [comment]
I've created #103 PR, but I will review the logic. I think there might be high resource utilization without a worker pool pattern. I will keep you posted

**Comment by wwoytenko** [comment]
@TCY16 

The large object dumper seg fault I hope will be fixed in #103 

Regarding this

> On a different note, I've also found that greenmask dump --data-only gives a schema collision error : "pg_dump: error: options -s/--schema-only and -a/--data-only cannot be used together"

I've opened issue #104 and initialized a PR #105 

The fix is expected to be published as v0.1.14 on Thursday (09.05.2024)

**Comment by wwoytenko** [comment]
Regarding --data-only workaround right now

I recommend creating dump as usual without --data-only.

But if you need to restore only data you can call restore --data-only.

**Comment by TCY16** [comment]
Thanks for the super quick turnaround! I'll post more issues if they come up 😄

**Comment by wwoytenko** [comment]
Published in 0.1.14. Let me know if everything is ok
